File: generate.java
// Time Complexity :O(n^2) where n is numRows;
// Space Complexity : O(n) n is the number of elements in the list. (Will it be O(1)? since the need to produce a list on line 12 for the output)
// Did this code successfully run on Leetcode :Yes
// Any problem you faced while coding this :No

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();
        if(numRows == 0) return result;
        
        for(int i =0;i<numRows;i++){
            List<Integer> list = new ArrayList<>();
            for(int j=0;j<=i;j++){
                if(j==0 || j == i){
                    list.add(1);
                }
                else{
                    list.add(result.get(i-1).get(j-1) + result.get(i-1).get(j));
                }
                
                
                
            }
            result.add(list);
            System.out.println(result);
        }
        
        return result;
    }
}

File: generate.js
// Time Complexity : O(n) where n is length of numRows
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    let result = [];
    result.push([1]);
    if(numRows === 1) return result;
    let a = 0;
    for(let i=1; i< numRows; i++) {
        let temp = [];
        temp.push(1);
        for(let j=1; j<i; j++) {
            temp.push(result[i-1][j] + result[i-1][j-1]);
        }
        temp.push(1);
        result.push(temp);
    }
    return result;
};